frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -204,20 +186,27 @@ export const DataPickerModal = ({
       });
     }
 
-    if (hasQuestions && hasNestedQueriesEnabled && models.includes("card")) {
+    if (
+      (hasQuestions || hasMetrics || hasModels) &&
+      hasNestedQueriesEnabled &&
+      (models.includes("card") ||
+        models.includes("dataset") ||
+        models.includes("metric"))
+    ) {
```

Comment:
To make the code a bit more self-documenting, you could define a variable with an informative name, maybe this:

```ts
const shouldShowCollectionsTab = (hasQuestions || hasMetrics || hasModels) &&
      hasNestedQueriesEnabled &&
      (models.includes("card") ||
        models.includes("dataset") ||
        models.includes("metric"));
```

And then do `if (shouldShowCollectionsTab) {`